doc/model.rst

```diff
@@ -0,0 +1,739 @@
+by the application and sent to the database as such: `model_name.tensor_name`.
+
+This is done simply through functions offered by the Model API:
+
+* Model.register_incoming_entity(incoming_entity)
+* Model.enable_key_prefixing()
+* Model.disable_key_prefixing()
+* Model.query_key_prefixing()
+
+-------
+Example
+-------
+We provide a producer/consumer example that demonstrates
+two producer models, with key prefixing enabled, that
+send tensor of the same name to a standard database. A
+consumer Model requests both tensors and displays the
+information using a SmartRedis logger to demonstrates
+successful use of Model key prefixing.
+
+During the example we will be creating four different files:
+
+1. producer_1.py : a Model producer application
+2. producer_2.py : a Model producer application
+3. consumer.py : a Model consumer application
+4. experiment.py : the Experiment driver script
+
+Producer_1 Application
+======================
+Begin by importing the necessary modules and initializing a SmartRedis
+Client:
+
+.. code-block:: python
+
+    from smartredis import Client
+    from smartredis import *
+    import numpy as np
+
+    # Initialize a Client
+    client = Client(cluster=False)
+
+Next, create a NumPy array to place in the database with the key name `tensor`:
+
+.. code-block:: python
+
+    # Create NumPy array
+    array = np.array([5, 6, 7, 8])
+    # Use SmartRedis client to place tensor in single sharded db
+    client.put_tensor("tensor", array)
+
+With key prefixing enabled, the tensor stored will be under key
+`producer_1.tensor`.
+
+Producer_2 Application
+======================
+Begin by importing the necessary modules and initializing a SmartRedis
+Client:
+
+.. code-block:: python
+
+    from smartredis import Client
+    from smartredis import *
+    import numpy as np
+
+    # Initialize a Client
+    client = Client(cluster=False)
+
+Next, create a NumPy array to place in the database with the key name `tensor`:
+
+.. code-block:: python
+
+    # Create NumPy array
+    array = np.array([1, 2, 3, 4])
+    # Use SmartRedis client to place tensor in single sharded db
+    client.put_tensor("tensor", array)
+
+With key prefixing enabled, the tensor stored will be under key
+`producer_2.tensor`.
+
+Consumer Application
+====================
+Next, request the inputted tensors from the producer
+applications within the consumer application.
+Begin by importing the necessary modules and initializing a SmartRedis
+Client:
+
+.. code-block:: python
+
+    from smartredis import Client
+    from smartredis import *
+    import numpy as np
+
+    # Initialize a Client
+    client = Client(cluster=False)
+
+SmartRedis offers the function, ``Client.set_data_source()`` that
+will add a prefix to the key name when using ``Client.get_tensor()``.
+
+.. code-block:: python
+
+    client.set_data_source("producer_1")
+    # Searching for key name: producer_1.tensor
```

Comment:
I would not use the word "searching", as that would be more appropriate for `poll` or `key_exists`. This is only retrieval.